src/java.management/share/classes/com/sun/jmx/remote/security/MBeanServerFileAccessController.java

```diff
@@ -300,16 +302,19 @@ private static Properties propertiesFromFile(String fname)
         }
     }
 
+    @SuppressWarnings("removal")
     private synchronized void checkAccess(AccessType requiredAccess, String arg) {
-        @SuppressWarnings("removal")
-        final AccessControlContext acc = AccessController.getContext();
-        @SuppressWarnings("removal")
-        final Subject s =
-            AccessController.doPrivileged(new PrivilegedAction<>() {
-                    public Subject run() {
-                        return Subject.getSubject(acc);
-                    }
-                });
+        final Subject s;
+        if (!SharedSecrets.getJavaLangAccess().allowSecurityManager()) {
+            s = Subject.current();
```

Comment:
We may not want to call `Subject.current()` here, as this may imply that we will support this functionality even if an SM is not enabled.

Reply:
I think we need @kevinjwalls or @dfuch to help advise on this.

Reply:
Right, this does not depend on the SM.   All we need to do is get the Subject.
This method implements the basic monitor (readonly) and control (readwrite) access.
accessMap maps identity String to Access, and the checkAccess() method here will check the Subject by using of its Principal names as keys in that map.

Reply:
Subject is stored in the RMIConnectionImpl: src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnectionImpl.java

(That is complicated by SubjectDelegation, which we deprecated for removal.  I have the PR out to remove it:
https://github.com/openjdk/jdk/pull/18025 )

makeClient in RMIJRMPServerImpl creates RMIConnectionImpl

..and RMIServerImpl.java has a doNewClient method calling that.  This is what takes a Credentials Object and deals withJMXAuthenticator to get an authenticated Subject.  None of this requires the SM.

Reply:
Yes - the JMX implementation stores and retrieve subjects in the AccessControlContext and then uses doPrivileged. Subject.doAs is not used in the JMX implementation. There are two different uses of Subject in JMX: 

1. one is a simplified role-based authentication/authorization at the default agent level. 
2. The other one is a permission check where different subjects can be granted different privileges in the policy file. 

The latter will go away since the SM is going away, but needs to be preserved until then.
The former we want to keep and needs to keep working (by changing the code to use callAs) even after the SM is gone.

Subject delegation allows an authenticated subject (1. above) to perform actions on behalf of a delegation subject, where the privileged granted by 2. are the intersection of the privileges of the two subjects.
@kevinjwalls is currently working on removing subject delegation.